### DISCUSSION TITLE
WIP & WIA Include only "InProgress" Items

### DISCUSSION THREAD

**Discussion by huserben** [nbrown02/ADO-Process-Behaviour-Chart#6] [Q&A] [answered]
Hi @nbrown02 

I just noticed that at least for the WIP and the Work Item Age, we filter for "StateCategory" eq "InProgress". I realized this when I looked at our board and it wasn't matching with what I was seeing in the PBC, because we have a few columns where the items are "Active" (InProgress) but after that some where the state is "Resolved" (StateCategory = Resolved).

I was wondering:
- Is this on purpose?
- If yes, why? (As I feel like it's "In progress until it's closed, right?)
- If no, should we adjust it?

I haven't checked the FlowViz template (but I think it's less of an issue there, unless maybe in the Daily WIP).

What do you think?

Cheers
Benji

**Comment by nbrown02** [comment, marked as answer]
Yeah it's a hard one as I often find teams who work with a resolved state aren't working on it, all about what you experience. The WIP chart in FlowViz works the same way so it's consistent but I could see why you'd want to change it in your context. You'd just need to edit the query to be _StateCategory%20ne%20%27Completed%27%20and%20StateCategory%20ne%20%27Proposed%27%20_

I think keep this discussion open for others to reply if it's an issue...i haven't had anything in the FlowViz template so far...

**Reply by huserben** [reply]
Yeah I was thinking that there might be a good reason for it. Was just confused at first when it was "not matching" my expectation ;-)

Thanks for clarifying

**Reply by huserben** [reply]
But I'm still a bit puzzled, maybe you can help me clarify where I have a gap in my thoughts :-)

Following is the WIA chart from Flow Viz:
![WIA](https://github.com/nbrown02/Process-Behaviour-Chart/assets/5486874/8a53eef9-1157-45cd-ac8b-12d327fc8ba0)

This matches with what we see on the board, the total WIA is 33.
In the X-Chart for Total Work Item Age, I see the following:

![TWIA X-Chart](https://github.com/nbrown02/Process-Behaviour-Chart/assets/5486874/00b79d08-d28c-439a-99a1-d4f93558e7cf)

The TWIA is 22 days. Which is quite a big gap, is there any explanation on what might be missing?

I used your adjustment statement from above, but somehow things don't add up. 🙃

**Reply by nbrown02** [reply]
I think I can see what the issue is, the work item age calculation is done in a column using DAX (and NOW for the current date/time) in FlowViz:
![image](https://github.com/nbrown02/Process-Behaviour-Chart/assets/29369962/3e63930a-40ec-4be4-9a34-2bf35cefb252)

Compared to Power Query (and using the DateValue column for the current date/time) in PBC (I've added resolved into the version I'm using in these screenshots):
![image](https://github.com/nbrown02/Process-Behaviour-Chart/assets/29369962/257cafb7-6fb0-4a05-ad47-882063bbb43a)

So the PBC template needs to be updated to align :)

**Reply by nbrown02** [reply]
@huserben try the latest and let me know if that fixes it...

**Reply by huserben** [reply]
Yes, the work item age now "matches" both what I see in Flow Viz and what I can see in Azure DevOps.

Thank you for the update!

**Reply by nbrown02** [reply]
Thank you for spotting it, quite a niche bug 😂

The FlowViz template has also been updated to now include the resolved state category :)

**Reply by huserben** [reply]
Oh, I might have found another "problem", but I'm not sure.

Now the PBC tells me I have a TWIA of 40 days (in my example). FlowViz tells me I have 33 days, and I have 7 items in progress. When I first checked I assumed it's simply because I did not update the FlowViz data locally yet (33 + 7  = 40 after all)...

But even now in my published report with both datasets updated this morning, the difference is there. I'm assuming it's some sort of different calculation where FlowViz takes another "cut-off" date than the "PBC", so in the PBC all items are +1 day older?

**Reply by huserben** [reply]
Oh, I spotted now that you updated FlowViz now...probably I should check that too before complaining 😆 
Let me double-check and get back to you. Question: did you "just" update the query or something else too? As I made some custom modification in flow viz it's easier if I don't have to "reload" the template.

**Reply by huserben** [reply]
Ok, even with the new FlowViz template, the data is not "matching", the PBC is still "one day ahead".

FlowViz: Total Work Item Age (manually summed up by me): 33
WIP: 7
PBC TWIA: 40

**Reply by nbrown02** [reply]
And I imagine this is due to the timing and usage of NOW in the DAX - the alternative I could try is mimic the calculation of days in progress using Power Query

**Reply by huserben** [reply]
Right - If I find some time later I can play around with it :)

I would assume the "day" makes more sense than "Now", or what would be your proposal. Which "end" to change? 😉

**Reply by huserben** [reply]
Sorry I hope I don't spam you too much. But, shouldn't we calculate Work Item Age like this:

> (Today - Day Started)  + 1

I see that in the PBC calculation, we add the "+1":
`= Table.TransformColumns(#"Changed Type2", {{"Custom", each _ + 1, type number}})`

But in the WorkItemAge table from FlowViz, we don't have it:
```
DaysInProgress = 
IF (
    'WorkItemsInProgress'[StateCategory] = "InProgress" || WorkItemsInProgress[StateCategory] = "Resolved",
    NOW () - 'WorkItemsInProgress'[InProgressDate]
)
```


What I did now is simply to add the "extra day":
```
DaysInProgress = 
IF (
    'WorkItemsInProgress'[StateCategory] = "InProgress" || WorkItemsInProgress[StateCategory] = "Resolved",
    (NOW () - 'WorkItemsInProgress'[InProgressDate]) + 1
)
```

Now that works and it's matching everywhere.

Does that make sense? Would it also be meaningful to propagate this change in FlowViz?

**Reply by nbrown02** [reply]
Ha this is how these things work! Invariably after you send a message an idea occurs :) 

That seems logical, does that fix it for you?

**Reply by huserben** [reply]
yeah, we're having alignment in Total Work Item Age now, so I'm happy 😄

**Reply by nbrown02** [reply]
Ok perfect - then I will get the main formula in FlowViz updated with that :)

**Reply by nbrown02** [reply]
All updated now :)

**Reply by huserben** [reply]
Great - I'll update FlowViz where I have it in use then. Thanks for the support Nick!
